************* Module app.test.test_utility
app/test/test_utility.py:2:0: C0301: Line too long (127/100) (line-too-long)
app/test/test_utility.py:23:0: C0301: Line too long (133/100) (line-too-long)
app/test/test_utility.py:28:0: C0304: Final newline missing (missing-final-newline)
app/test/test_utility.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/test/test_utility.py:4:0: E0401: Unable to import 'src.connection_utility' (import-error)
app/test/test_utility.py:5:0: E0401: Unable to import 'sqlalchemy' (import-error)
app/test/test_utility.py:6:0: E0401: Unable to import 'pandas' (import-error)
app/test/test_utility.py:8:0: C0116: Missing function or method docstring (missing-function-docstring)
************* Module app.test.test_manager
app/test/test_manager.py:2:0: C0301: Line too long (127/100) (line-too-long)
app/test/test_manager.py:14:0: C0304: Final newline missing (missing-final-newline)
app/test/test_manager.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/test/test_manager.py:6:0: E0401: Unable to import 'sqlalchemy' (import-error)
app/test/test_manager.py:7:0: E0401: Unable to import 'pandas' (import-error)
app/test/test_manager.py:9:0: C0116: Missing function or method docstring (missing-function-docstring)
app/test/test_manager.py:6:0: C0411: third party import "sqlalchemy.create_engine" should be placed before first party import "app.src.connection_manager.main"  (wrong-import-order)
app/test/test_manager.py:7:0: C0411: third party import "pandas" should be placed before first party import "app.src.connection_manager.main"  (wrong-import-order)
app/test/test_manager.py:6:0: W0611: Unused create_engine imported from sqlalchemy (unused-import)
app/test/test_manager.py:7:0: W0611: Unused pandas imported as pd (unused-import)
************* Module app.src.connection_utility
app/src/connection_utility.py:23:0: C0301: Line too long (147/100) (line-too-long)
app/src/connection_utility.py:26:0: C0301: Line too long (194/100) (line-too-long)
app/src/connection_utility.py:29:0: C0301: Line too long (147/100) (line-too-long)
app/src/connection_utility.py:39:0: W0311: Bad indentation. Found 9 spaces, expected 8 (bad-indentation)
app/src/connection_utility.py:78:0: C0303: Trailing whitespace (trailing-whitespace)
app/src/connection_utility.py:88:0: C0304: Final newline missing (missing-final-newline)
app/src/connection_utility.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/src/connection_utility.py:2:0: E0401: Unable to import 'yaml' (import-error)
app/src/connection_utility.py:21:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_utility.py:22:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
app/src/connection_utility.py:21:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
app/src/connection_utility.py:44:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_utility.py:45:4: W0621: Redefining name 'connections' from outer scope (line 87) (redefined-outer-name)
app/src/connection_utility.py:47:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
app/src/connection_utility.py:53:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_utility.py:53:22: W0621: Redefining name 'connections' from outer scope (line 87) (redefined-outer-name)
app/src/connection_utility.py:70:12: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
app/src/connection_utility.py:3:0: C0411: standard import "os" should be placed before third party import "yaml" (wrong-import-order)
************* Module app.src.cli
app/src/cli.py:47:0: C0301: Line too long (102/100) (line-too-long)
app/src/cli.py:48:0: C0301: Line too long (103/100) (line-too-long)
app/src/cli.py:50:0: C0301: Line too long (135/100) (line-too-long)
app/src/cli.py:52:0: C0301: Line too long (125/100) (line-too-long)
app/src/cli.py:53:0: C0303: Trailing whitespace (trailing-whitespace)
app/src/cli.py:96:0: C0305: Trailing newlines (trailing-newlines)
app/src/cli.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/src/cli.py:1:0: E0401: Unable to import 'sqlalchemy' (import-error)
app/src/cli.py:41:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/cli.py:81:8: C0415: Import outside toplevel (main_call.return_engine) (import-outside-toplevel)
app/src/cli.py:88:15: W0718: Catching too general exception Exception (broad-exception-caught)
app/src/cli.py:4:0: C0411: standard import "argparse" should be placed before third party import "sqlalchemy.text" and local imports "connection_manager.main", "main_call.interactive" (wrong-import-order)
app/src/cli.py:5:0: C0411: standard import "os" should be placed before third party import "sqlalchemy.text" and local imports "connection_manager.main", "main_call.interactive" (wrong-import-order)
app/src/cli.py:6:0: C0411: standard import "sys" should be placed before third party import "sqlalchemy.text" and local imports "connection_manager.main", "main_call.interactive" (wrong-import-order)
************* Module app.src.connection_manager
app/src/connection_manager.py:66:146: C0303: Trailing whitespace (trailing-whitespace)
app/src/connection_manager.py:66:0: C0301: Line too long (146/100) (line-too-long)
app/src/connection_manager.py:78:0: C0301: Line too long (127/100) (line-too-long)
app/src/connection_manager.py:84:0: C0301: Line too long (106/100) (line-too-long)
app/src/connection_manager.py:95:0: C0301: Line too long (135/100) (line-too-long)
app/src/connection_manager.py:101:0: C0301: Line too long (149/100) (line-too-long)
app/src/connection_manager.py:111:0: C0301: Line too long (131/100) (line-too-long)
app/src/connection_manager.py:113:0: C0301: Line too long (129/100) (line-too-long)
app/src/connection_manager.py:114:0: C0301: Line too long (123/100) (line-too-long)
app/src/connection_manager.py:115:0: C0301: Line too long (135/100) (line-too-long)
app/src/connection_manager.py:116:0: C0301: Line too long (123/100) (line-too-long)
app/src/connection_manager.py:117:0: C0301: Line too long (123/100) (line-too-long)
app/src/connection_manager.py:118:0: C0301: Line too long (135/100) (line-too-long)
app/src/connection_manager.py:119:0: C0301: Line too long (161/100) (line-too-long)
app/src/connection_manager.py:147:0: C0304: Final newline missing (missing-final-newline)
app/src/connection_manager.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/src/connection_manager.py:1:0: E0401: Unable to import 'yaml' (import-error)
app/src/connection_manager.py:3:0: E0401: Unable to import 'tabulate' (import-error)
app/src/connection_manager.py:24:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_manager.py:26:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
app/src/connection_manager.py:31:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_manager.py:32:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
app/src/connection_manager.py:36:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_manager.py:42:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_manager.py:60:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_manager.py:76:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/connection_manager.py:84:12: C0103: Variable name "ConnType" doesn't conform to snake_case naming style (invalid-name)
app/src/connection_manager.py:85:32: C0321: More than one statement on a single line (multiple-statements)
app/src/connection_manager.py:85:32: C0103: Variable name "ConnType" doesn't conform to snake_case naming style (invalid-name)
app/src/connection_manager.py:86:34: C0321: More than one statement on a single line (multiple-statements)
app/src/connection_manager.py:86:34: C0103: Variable name "ConnType" doesn't conform to snake_case naming style (invalid-name)
app/src/connection_manager.py:87:34: C0321: More than one statement on a single line (multiple-statements)
app/src/connection_manager.py:87:34: C0103: Variable name "ConnType" doesn't conform to snake_case naming style (invalid-name)
app/src/connection_manager.py:88:34: C0321: More than one statement on a single line (multiple-statements)
app/src/connection_manager.py:88:34: C0103: Variable name "ConnType" doesn't conform to snake_case naming style (invalid-name)
app/src/connection_manager.py:76:0: R0912: Too many branches (17/12) (too-many-branches)
app/src/connection_manager.py:76:0: R0915: Too many statements (62/50) (too-many-statements)
app/src/connection_manager.py:2:0: C0411: standard import "os" should be placed before third party import "yaml" (wrong-import-order)
************* Module app.src.main_call
app/src/main_call.py:1:0: C0301: Line too long (117/100) (line-too-long)
app/src/main_call.py:41:0: C0301: Line too long (173/100) (line-too-long)
app/src/main_call.py:47:0: C0301: Line too long (164/100) (line-too-long)
app/src/main_call.py:50:0: C0305: Trailing newlines (trailing-newlines)
app/src/main_call.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/src/main_call.py:2:0: E0401: Unable to import 'sqlalchemy' (import-error)
app/src/main_call.py:3:0: E0401: Unable to import 'pandas' (import-error)
app/src/main_call.py:5:0: E0401: Unable to import 'yaml' (import-error)
app/src/main_call.py:8:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/main_call.py:18:0: C0116: Missing function or method docstring (missing-function-docstring)
app/src/main_call.py:22:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
app/src/main_call.py:26:4: C0103: Variable name "CONNECTION_STRING" doesn't conform to snake_case naming style (invalid-name)
app/src/main_call.py:41:4: C0103: Constant name "string" doesn't conform to UPPER_CASE naming style (invalid-name)
app/src/main_call.py:2:0: C0411: third party import "sqlalchemy.create_engine" should be placed before local import "connection_utility.make_string" (wrong-import-order)
app/src/main_call.py:3:0: C0411: third party import "pandas" should be placed before local import "connection_utility.make_string" (wrong-import-order)
app/src/main_call.py:4:0: C0411: standard import "os" should be placed before third party imports "sqlalchemy.create_engine", "pandas" and local import "connection_utility.make_string" (wrong-import-order)
app/src/main_call.py:5:0: C0411: third party import "yaml" should be placed before local import "connection_utility.make_string" (wrong-import-order)
app/src/main_call.py:1:0: R0801: Similar lines in 2 files
==app.src.connection_manager:[7:24]
==app.src.connection_utility:[5:21]
conn_home = os.getenv('conn_home')
# print(f"conn_home: {conn_home}")

# Define the path for the connections file
connections_file = os.path.join(conn_home, 'connections.yaml')
# print(f"connections_file: {connections_file}")

# Check if the file exists
if os.path.exists(connections_file):
    # print("File exists.")
    pass # do nothing
else:
    print("File does not exist.")

# @trace_func
def make_string(conn_det): (duplicate-code)

-----------------------------------
Your code has been rated at 3.86/10

